spring-cloud-netflix-core/src/test/java/OkHttpClientConfigurationTests.java

```diff
@@ -0,0 +1,160 @@
+ *  * limitations under the License.
+ *
+ */
+
+import feign.Client;
+import okhttp3.ConnectionPool;
+import okhttp3.OkHttpClient;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.X509TrustManager;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockingDetails;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.commons.httpclient.DefaultOkHttpClientConnectionPoolFactory;
+import org.springframework.cloud.commons.httpclient.DefaultOkHttpClientFactory;
+import org.springframework.cloud.commons.httpclient.OkHttpClientConnectionPoolFactory;
+import org.springframework.cloud.commons.httpclient.OkHttpClientFactory;
+import org.springframework.cloud.netflix.feign.FeignClient;
+import org.springframework.cloud.netflix.feign.ribbon.LoadBalancerFeignClient;
+import org.springframework.cloud.netflix.ribbon.okhttp.OkHttpLoadBalancingClient;
+import org.springframework.cloud.netflix.ribbon.support.RibbonRequestCustomizer;
+import org.springframework.cloud.netflix.zuul.EnableZuulProxy;
+import org.springframework.cloud.netflix.zuul.filters.route.RibbonCommandContext;
+import org.springframework.cloud.netflix.zuul.filters.route.okhttp.OkHttpRibbonCommand;
+import org.springframework.cloud.netflix.zuul.filters.route.okhttp.OkHttpRibbonCommandFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockingDetails;
+
+/**
+ * @author Ryan Baxter
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = OkHttpClientConfigurationTestApp.class, value = {"feign.okhttp.enabled: true",
+		"spring.cloud.httpclientfactories.ok.enabled: true", "ribbon.eureka.enabled = false", "ribbon.okhttp.enabled: true"})
+@DirtiesContext
+public class OkHttpClientConfigurationTests {
+
+	@Autowired
+	OkHttpClientFactory okHttpClientFactory;
+
+	@Autowired
+	OkHttpClientConnectionPoolFactory connectionPoolFactory;
+
+	@Autowired
+	LoadBalancerFeignClient feignClient;
+
+	@Autowired
+	OkHttpRibbonCommandFactory okHttpRibbonCommandFactory;
+
+	@Test
+	public void testFactories() {
+		assertTrue(OkHttpClientConnectionPoolFactory.class.isInstance(connectionPoolFactory));
```

Comment:
I'd prefer `Assertions.assertThat(connectionPoolFactory).isInstanceOf(OkHttpClientConnectionPoolFactory.class)`. That way, if it fails, we have a decent message :-)